### ISSUE TITLE
Add syntax of custom IDs for headings

### ISSUE THREAD

**Issue by orelby** [GitbookIO/gitbook#382]
Lots of UTF-8 characters aren't supported in various ebooks' table of contents (or just calibre generated ebooks?) - meaning some aren't presented in the reader (or are gibberish) and all non-English (even question marks) don't link to their corresponding chapter/section.

One could just remove the automated ID generation (it should be configurable), but what about HTML-based ebooks?

Another, non contradictory approach would be to **add configuration for disabling automatic ID** generation in certain places (html/ebook-formats/all).

**tl;dr** The syntax could be like that:

```
Heading 1 {#heading1}
========

## Heading 2 {#heading2}
```
